qiskit/circuit/quantumcircuit.py

```diff
@@ -955,30 +958,41 @@ def compose(
                 )
             edge_map.update(zip(other.clbits, dest.cbit_argument_conversion(clbits)))
 
+        register_map = {}
+
+        def map_register_to_dest(theirs, register_map):
+            """Map the target's registers to suitable equivalents in the destination, adding an
+            extra one if there's no exact match."""
+            if theirs.name in register_map:
+                return register_map[theirs.name]
+            mapped_bits = [edge_map[bit] for bit in theirs]
+            for ours in dest.cregs:
+                if mapped_bits == list(ours):
+                    mapped_theirs = ours
+                    break
+            else:
+                mapped_theirs = ClassicalRegister(bits=[edge_map[bit] for bit in theirs])
```

Comment:
I see this came from the original code, but should we replace `[edge_map[bit] for bit in theirs]` with `mapped_bits` since we've already computed this list?